sql/src/main/java/io/crate/metadata/Reference.java

```diff
@@ -56,6 +57,12 @@
     private final boolean nullable;
     private final boolean columnStoreDisabled;
 
+    @Nullable
+    private final String formattedDefaultExpression;
```

Comment:
Does the `Reference` really also need both, the `defaultExpression` and `formattedDefaultExpression` ?

If this is due to the construction of the References in `DocIndexMetaData` - couldn't we parse the expressions eagerly there? Given that they cannot have dependencies on other references that should be possible.

Reply:
Yeap, it should be possible. 
I also thought about it but decided to keep it this way to re-use the resources (parser etc) already created (lazily) for the generated columns.

I will change it though - I like the idea to get rid of the formatted expression.